### PR TITLE
Make the deployment of an IngressClass optional with a default to true

### PR DESCRIPTION
**Pull request by Rumbles**
Add a variable to the k8s operator values which defaults to true, and use that to decide whether to install the IngressClass.

Links to #14152
Fixes tailscale/tailscale#14152

### CODE REVIEW

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thank you for the PR @Rumbles , I've added a nit to restructure the values, apart from that it makes sense to me.

Could you also link the issue in your commit message to pass our CI issue checker? 
i.e `Fixes tailscale/tailscale#14152` - see other commits in this repo

File: cmd/k8s-operator/deploy/chart/values.yaml [hunk 1]
```diff
@@ -54,6 +54,8 @@ operatorConfig:
   # - name: EXTRA_VAR2
   #   value: "value2"
 
+  # In the case that you already have a tailscale ingressclass in your cluster (or vcluster), set this to false
+  createIngressClass: true
```
[the remark this answers is not in the document]
Comment (Rumbles) [reply cmd/k8s-operator/deploy/chart/values.yaml]: Hi @irbekrm thanks for the review, I have updated my PR

[the remark this answers is not in the document]
Comment (irbekrm) [reply cmd/k8s-operator/deploy/chart/values.yaml]: Thanks @Rumbles the code change looks good, could you just please squash this into a single signed commit and add that `Fixes <issue>` or `Updates <issue>` line to the actual commit message?
i.e like https://github.com/tailscale/tailscale/commit/00517c8189569171560c073cd983164ff7735e69

Apart from that looks good to go in

[the remark this answers is not in the document]
Comment (Rumbles) [reply cmd/k8s-operator/deploy/chart/values.yaml]: Hi, squashed and amended @irbekrm

[the remark this answers is not in the document]
Comment (Rumbles) [reply cmd/k8s-operator/deploy/chart/values.yaml]: Sorry I misunderstood that part of the message, commit updated

**Review by irbekrm — APPROVED** [review]
Thank you @Rumbles !

**Comment by irbekrm** [comment]
@Rumbles `1.77.119` Tailscale dev helm chart will have this change

**Comment by Rumbles** [comment]
Thanks for the info @irbekrm and the review/merge, I'm looking forward to trying it out :)

**Comment by irbekrm** [comment]
> I'm looking forward to trying it out :)

Any feedback if you run into further issues is welcome, we might want to add some docs for running on vcluster at some point

**Pull request merged**